modules/lang-painless/src/main/java/org/elasticsearch/painless/node/ECapturingFunctionRef.java

```diff
@@ -63,51 +59,53 @@ Output analyze(ClassNode classNode, ScriptRoot scriptRoot, Scope scope, Input in
                     "not a statement: capturing function reference [" + variable + ":"  + call + "] not used"));
         }
 
-        FunctionRef ref = null;
-
         Output output = new Output();
 
-        captured = scope.getVariable(location, variable);
+        Variable captured = scope.getVariable(location, variable);
         if (input.expected == null) {
+            String defReferenceEncoding;
             if (captured.getType() == def.class) {
                 // dynamic implementation
-                defPointer = "D" + variable + "." + call + ",1";
+                defReferenceEncoding = "D" + variable + "." + call + ",1";
```

Comment:
Add TODO, let's pull this encoding creation out into something that makes the arguments clear.

Reply:
https://github.com/elastic/elasticsearch/issues/55543